internal/data/metric.go

```diff
@@ -184,12 +179,17 @@ func (rm ResourceMetrics) InstrumentationLibraryMetrics() []InstrumentationLibra
 func (rm ResourceMetrics) SetInstrumentationLibraryMetrics(s []InstrumentationLibraryMetrics) {
 	rm.initInternallIfNeeded()
 	rm.pimpl.instrumentationLibraryMetrics = s
-	// We don't update the orig slice because this may be called multiple times.
-	rm.pimpl.sliceChanged = true
-}
 
-func (rm ResourceMetrics) getOrig() *otlpmetrics.ResourceMetrics {
-	return rm.orig
+	if len(rm.pimpl.instrumentationLibraryMetrics) == 0 {
+		rm.orig.InstrumentationLibraryMetrics = nil
+		return
+	}
+
+	// Reconstruct the slice because we don't know what elements were removed/added.
```

Comment:
Same here, would be good to add a TODO to reuse `rm.orig`

Reply:
Done.